lib/ui/pages/web_view/web_view_page.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:flutter/material.dart';
+import 'package:webview_flutter/webview_flutter.dart';
+
+class WebViewPage extends StatelessWidget {
+  final String url;
+
+  WebViewPage({Key key, this.url})
+      : assert(url != null),
```

Comment:
co do assercji to wczoraj była u nas o tym dyskusja i stwierdzono, że nigdzie nie będziemy tego dawać. Cytuje co przemówiło za tym:


>Jak dla mnie powinny być wszędzie, gdzie masz @required, albo nigdzie. Bo żeby wiedzieć czy gdzieś dalej jest to sprawdzone musisz wiedzieć jak kod dalej wygląda, edytować w przypadku zmian itd. Ja osobiście nie używałbym tego nigdzie z dwóch powodów.
>
>1. Działają tylko w debugu.
>2. Taka rzecz jak null nie stnieje i w przypadku braku wartości korzysta się Optional/Option.

co myślisz?

Reply:
hmm... zastanawiam się mocno, po wstępnym zastanowieniu się rozważyłbym niekorzystanie z asercji z uwagi na to że być może fałszywie chronią przed wadliwym stanem przed którym bronią w debugu, ale ten wadliwy stan może mieć miejsce w produkcyjnej wersji i przez to aplikacja może nie być przygotowana na taki stan. Ale zastanawiam się w takim razie, czy w takim razie nie powinno się korzystać z żadnych asercji? Wasza dyskusja dotyczyła wyłącznie sprawdzenia czy jest różne od null czy ogólnie wszystkich asercji? Z drugiej strony te asercje mają tę zaletę, że ułatwiają development bo pokazują to w jaki sposób klasa została zaprojektowana i w jaki sposób należy z niej korzystać. Generalnie widzę, że asercje raczej są szeroko stosowane przez google 🤔

Reply:
w tym konkretnym przykładzie chodziło o asercję na optional named parameter oznaczone adnotacją @required, ale też o nulle i używanie tego Optional w tym przypadku. Szczerze mówiąc to jeszcze nie sprawdzałem jak to miałoby wyglądać. Też widziałem, że google namiętnie używa assercji. Możemy ustalić, że nie używamy asercji na optional named parametrach oznaczonych adnotacją @required.

Reply:
generalnie to pracują nad nienullowalnymi i nullowalnymi typami jak w kotlinie także niedługo problem będzie solved out of the box :D